### PR TITLE
Onedotoh

### PR DESCRIPTION
**Pull request by nickdesaulniers**
review? @reqshark

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -226,6 +226,24 @@ socket.ipv6(true);
...
+
+### socket.rmchan()
+
+*(Function, param: String)*: Allows for sub sockets to remove channel filters.
+Not applicable to non sub sockets.
+
+If you unsubscribe from the default channel, `''`, without subscribing to any
+new channels, your sub socket will stop receiving messages.
+
+### socket.rmchan()
```
Comment (reqshark) [inline README.md]: much copy-paste i think, extra `socket.rmchan()` line

File: README.md [hunk 2]
```diff
@@ -226,6 +226,24 @@ socket.ipv6(true);
 console.log(socket.ipv6()); // true
 ```
 
+### socket.chan()
```
Comment (reqshark) [inline README.md]: can we put:
### socket.chan(Array)

File: README.md [hunk 3]
```diff
@@ -226,6 +226,24 @@ socket.ipv6(true);
...
+
+*(Function, param: String)*: Allows for sub sockets to filter messages based on
+a prefix. Not applicable to non sub sockets.
+
+By default, all sub sockets are subscribed to the `''` channel.  Once you opt
+in to filtering on a channel, you are unsubscribed from `''`.
+
+### socket.rmchan()
```
Comment (reqshark) [inline README.md]: #### socket.rmchan(Strings)

File: README.md [hunk 4]
```diff
@@ -226,6 +226,24 @@ socket.ipv6(true);
...
+
+*(Function, param: String)*: Allows for sub sockets to remove channel filters.
```
Comment (reqshark) [inline README.md]: maybe something like:
*(Function, param: String, variable arity)*

File: README.md [hunk 5]
```diff
@@ -226,6 +226,24 @@ socket.ipv6(true);
 console.log(socket.ipv6()); // true
 ```
 
+### socket.chan()
+
+*(Function, param: String)*: Allows for sub sockets to filter messages based on
```
Comment (reqshark) [inline README.md]: *(Function, param: Array)*

**Comment by nickdesaulniers** [comment]
ready for rereview @reqshark

**Comment by reqshark** [comment]
ok cool thanks that looks better. Those descriptive summaries are on point!

sorry also can we move those two methods above `socket.options`? 

lets put them right below `socket.connect(addr)` or `.close()` since subscriptions follow directly from endpoint connect/bind

also I forgot about the ability to pass channels as options when we init the socket.

we could add an option reference, maybe with the `#` hashtag scroll link shortcutter: 

18oi_-z7z2-3000x3000

so a `chan` property on the opts [accepts an array](https://github.com/nickdesaulniers/node-nanomsg/blob/master/lib/index.js#L106)... that was your idea btw :+1:

**Comment by nickdesaulniers** [comment]
ready for rereview @reqshark

**Pull request merged**